File: loop_transformations/sources/fuse.3.f90
! @@name:       fuse.3
! @@type:       F-free
! @@operation:  compile
! @@expect:     success
! @@version:    omp_6.0
subroutine func(n, A, B, C, D)
  implicit none
  integer,intent(in)  :: n
  real,   intent(out) :: A(n), B(n), C(n), D(n)
  integer        :: i, j, k, l 
  real, external :: e, f, g, h

  !$omp fuse looprange(2,2)
    do i = 1,n
      A(i) = e(i)
    end do
    do j = 1,n
      B(j) = f(j)
    end do
    do k = 1,n
      C(k) = g(k)
    end do
    do l = 1,n
      D(l) = h(l)
    end do
  !$omp end fuse
end subroutine

subroutine func_equivalent(n, A,  B,  C,  D)
  implicit none
  integer,intent(in)  :: n
  real,   intent(out) :: A(n), B(n), C(n), D(n)
  integer        :: i, jk, l 
  real, external :: e, f, g, h

  do i = 1,n
    A(i) = e(i)
  end do

  do jk = 1,n
    B(jk) = f(jk)
    C(jk) = g(jk)
  end do

  do l=1,n
    D(l) = h(l)
  end do
end subroutine

File: loop_transformations/sources/fuse.3.c
/*
* @@name:	fuse.3
* @@type:	C
* @@operation:	compile
* @@expect:	success
* @@version:	omp_6.0
*/
float e(int);
float f(int);
float g(int);
float h(int);

void func(int n, float* A, float* B, float* C, float* D)
{
  #pragma omp fuse looprange(2,2)
  {
    for (int i = 0; i < n; ++i)
      A[i] = e(i);
    for (int j = 0; j < n; ++j)
      B[j] = f(j);
    for (int k = 0; k < n; ++k)
      C[k] = g(k);
    for (int l = 0; l < n; ++l)
      D[l] = h(l);
  }
}

void func_equivalent(int n, float* A, float* B, float* C, float* D)
{
  for (int i  = 0;  i < n; ++i)
    A[i] = e(i);
  for (int jk = 0; jk < n; ++jk) {
    B[jk] = f(jk);
    C[jk] = g(jk);
  }
  for (int l  = 0;  l < n; ++l)
    D[l] = h(l);
}
